pypit/calibrations.py

```diff
@@ -270,138 +313,164 @@ def get_pixflatnrm(self):
-
-            # Load via master, as desired
-            if not self.traceSlits.master():
-                # Build the trace image first
-                trace_image_files = arsort.list_of_files(self.fitstbl, 'trace', self.sci_ID)
-                Timage = traceimage.TraceImage(trace_image_files,
-                                               spectrograph=self.spectro_class.spectrograph,
-                                               settings=self.settings, det=self.det,
-                                               datasec_img=self.datasec_img)
-                mstrace = Timage.process(bias_subtract=self.msbias, trim=self.settings['reduce']['trim'],
-                                         apply_gain=True)
-                # Load up and get ready
-                self.traceSlits.mstrace = mstrace
-                _ = self.traceSlits.make_binarr()
-                # Now we go forth
-                self.tslits_dict = self.traceSlits.run(arms=True)
-                # No slits?
-                if self.tslits_dict is None:
-                    return None, None
-                # QA
-                if self.write_qa:
-                    self.traceSlits._qa()
-                # Save to disk
-                if self.save_masters:
-                    # Master
-                    self.traceSlits.save_master()
-            else:
-                self.tslits_dict = self.traceSlits._fill_tslits_dict()
-            # Save in calib
-            self.calib_dict[self.setup]['trace'] = self.tslits_dict
+            self.tslits_dict = None
+            self.maskslits = None
+            return self.tslits_dict, self.maskslits
 
-        ###############################################################################
-        # Initialize maskslits
-        nslits = self.tslits_dict['lcen'].shape[1]
-        self.maskslits = np.zeros(nslits, dtype=bool)
+        # Check internals
+        self._chk_set(['setup', 'det', 'sci_ID', 'par'])
 
-        # Return
+        # Return already generated data
+        if 'trace' in self.calib_dict[self.setup].keys():
+            self.tslits_dict = self.calib_dict[self.setup]['trace']
+            self.maskslits = np.zeros(self.tslits_dict['lcen'].shape[1], dtype=bool)
+            return self.tslits_dict, self.maskslits
+                
+        # Instantiate (without mstrace)
+        self.traceSlits = traceslits.TraceSlits(None, self.pixlocn, par=self.par['slits'],
+                                                det=self.det, setup=self.setup,
+                                                directory_path=self.master_dir,
+                                                mode=self.par['masters'], binbpx=self.msbpm)
+
+        # Load via master, as desired
+        if not self.traceSlits.master():
+            # Build the trace image first
+            trace_image_files = arsort.list_of_files(self.fitstbl, 'trace', self.sci_ID)
+            Timage = traceimage.TraceImage(self.spectrograph,
```

Comment:
should this be called `traceImage` instead of `Timage` to keep with our standard?

Reply:
Done.